File: src/8.0.0/items/mediastorage/mediastorageprovidercontext.ts
import { Guid, GuidValue, MediaPickerStorageProviderContext } from "@omnia/fx-models";


//***************************************************************************
// TODO - Move this file somewhere under ./client folder
//***************************************************************************


//***************************************************************************
// To upload media to this storage from client-side
// Create a $outputname$Context object and put it in <omfx-media-picker providerContext={here} ... >
//***************************************************************************


export class $outputname$Context extends MediaPickerStorageProviderContext {
    readonly omniaServiceId: GuidValue = Your-web-service-id;
    readonly storageProviderContextId: GuidValue = new Guid("$guid1$");

    //***************************************************************************
    //This is just a test property for media context that can be used to allocate images or authorization
    //Remove or Update this property based on your needs
    //***************************************************************************
    
    businessProfileId?: GuidValue;

    //***************************************************************************
    //This is just a test property for media context that can be used to allocate images or authorization
    //Remove or Update this property based on your needs
    //***************************************************************************
    myItemId: number;

    constructor(myItemId: number, businessProfileId: GuidValue = null) {
        super();

        this.businessProfileId = businessProfileId;
        this.myItemId = myItemId;
    }
}

File: src/8.0.0/items/mediastorage/mediastorageprovidercontext.cs
using Newtonsoft.Json;
using Omnia.Fx.Models.MediaPicker;
using System;


namespace $namespace$
{

    //***************************************************************************
    // To upload media to this storage from server-side
    // Create a $outputname$Context object and use it within Omnia.Fx.MediaPicker.IMediaPickerService
    // e.g. mediaPickerService.AddNewImageAsync(...., providerContext: here )
    //***************************************************************************

    public class $outputname$Context : MediaPickerStorageProviderContext
    {
        public override Guid OmniaServiceId { get; protected set; } = new Guid(Your-web-service-id);
        public override Guid StorageProviderContextId { get; protected set; } = new Guid("$guid1$");



        //***************************************************************************
        // This is just a test property for media context that can be used to allocate images or authorization
        // Remove or Update this property based on your needs
        //***************************************************************************
        [JsonProperty]
        public int MyItemId { get; private set; }

        //***************************************************************************
        // This is a test property for media context that can be used to allocate images or authorization
        // // Remove or Update this property based on your needs
        //***************************************************************************
        [JsonProperty]
        public Guid? BusinessProfileId { get; private set; } 

        public $outputname$Context()
        {

        }

        public $outputname$Context(int myItemId, Guid? businessProfileId)
        {
            this.MyItemId = myItemId;
            this.BusinessProfileId = businessProfileId;
        }
    }
}
